### DISCUSSION TITLE
Authenticating an external service to use Concourse API

### DISCUSSION THREAD

**Discussion by chivince** [concourse/concourse#8275] [Help & Support] [answered]
Hello folks,

I would like to create a service that could use Concourse API, to get the pipelines and their statuses.
Therefore, I'm looking for a way to do such queries directly, without using fly, but I don't know how to authenticate the service.
Do you know how to achieve it ?

Thanks a lot !

**Comment by suhlig** [comment, marked as answer]
If you insist on _not_ using fly, [go-concourse](https://github.com/concourse/concourse/blob/master/go-concourse/README.md) may be the way to go. I am not aware of a example, but you could look at [how fly login works](https://github.com/concourse/concourse/blob/master/fly/commands/login.go) and build something similar.

But I think your request to _"get the pipelines and their statuses"_ may be much simpler to achieve with e.g. `fly builds --json`.

**Reply by chivince** [reply]
Thank you so much for your answer !

Actually, I found another topic addressing exactly what I wanted to do : https://github.com/concourse/concourse/discussions/6241#discussioncomment-120436
I just tried it, and using fly credentials, I'm able to get a token for a specific user, and then use it to access the API !

**Reply by tarunbiswas69609-svg** [reply]
F this like you hacked
